### PR TITLE
feat: [CODE-1181]: Support harness stage in status check

### PR DESCRIPTION
**Pull request by abhinav-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56331)

### CODE REVIEW

**Comment by abhinav-harness** [comment]
trigger uts

**Comment by abhinav-harness** [comment]
trigger uts

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/ci/CIBuildStatusPushTask.java [hunk 1]
```diff
@@ -36,8 +36,9 @@ public CIBuildStatusPushTask(DelegateTaskPackage delegateTaskPackage, ILogStream
 
   @Override
   public DelegateResponseData run(TaskParameters parameters) {
-    if (((CIBuildPushParameters) parameters).commandType == CIBuildPushTaskType.STATUS) {
-      CIBuildStatusPushParameters ciBuildStatusPushParameters = (CIBuildStatusPushParameters) parameters;
+    if (((io.harness.delegate.task.ci.CIBuildPushParameters) parameters).commandType == CIBuildPushTaskType.STATUS) {
```
Comment (shubham149) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/ci/CIBuildStatusPushTask.java:39]: Nit: Revert this.

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/ci/CIBuildStatusPushTask.java [hunk 2]
```diff
@@ -49,7 +50,7 @@ public DelegateResponseData run(TaskParameters parameters) {
     return BuildStatusPushResponse.builder().status(Status.ERROR).build();
   }
 
-  private GitStatusCheckParams convertParams(CIBuildStatusPushParameters params) {
+  private GitStatusCheckParams convertParams(io.harness.delegate.task.ci.CIBuildStatusPushParameters params) {
```
Comment (shubham149) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/ci/CIBuildStatusPushTask.java:53]: Use CIBuildStatusPushParameters

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/ci/CIBuildPushParameters.java [hunk 3]
```diff
@@ -42,13 +43,22 @@ public abstract class CIBuildPushParameters implements TaskParameters, Execution
   private String token; //  TODO it will come via bitbucket/gitlab  connector details
   private String userName; //  TODO it will come via bitbucket/gitlab connector details
   private ConnectorDetails connectorDetails; // Use connectorDetails to retrieve all information
-  private GitSCMType gitSCMType;
+  private io.harness.delegate.task.ci.GitSCMType gitSCMType;
```
Comment (shubham149) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/ci/CIBuildPushParameters.java:46]: Ditto

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/ci/CIBuildStatusPushParameters.java [hunk 4]
```diff
@@ -15,18 +15,20 @@
 
 @Value
 @EqualsAndHashCode(callSuper = false)
-public class CIBuildStatusPushParameters extends CIBuildPushParameters {
+public class CIBuildStatusPushParameters extends io.harness.delegate.task.ci.CIBuildPushParameters {
```
Comment (shubham149) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/ci/CIBuildStatusPushParameters.java:18]: ditto

File: 970-ng-commons/src/main/java/io/harness/code/HarnessCodePayload.java [hunk 5]
```diff
@@ -28,10 +27,25 @@ public class HarnessCodePayload {
   public static class Payload {
     String version;
     CheckPayloadKind kind;
-    JSONObject data;
+    HarnessPayloadData data;
```
Comment (shubham149) [inline 970-ng-commons/src/main/java/io/harness/code/HarnessCodePayload.java:30]: can this cause any backward compatibility issue?

Comment (abhinav-harness) [reply 970-ng-commons/src/main/java/io/harness/code/HarnessCodePayload.java:30]: its always sent through del

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/ci/CIBuildStatusPushTask.java (hunk 2 above)
Comment (abhinav-harness) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/ci/CIBuildStatusPushTask.java:53]: intellij is doing for me :( 
I am not able to not do it :(

**Comment by abhinav-harness** [comment]
trigger uts

**Comment by hitesharinga** [comment]
trigger messagecheck

File: 970-ng-commons/src/main/java/io/harness/code/HarnessCodePayload.java (hunk 5 above)
Comment (abhinav-harness) [reply 970-ng-commons/src/main/java/io/harness/code/HarnessCodePayload.java:30]: tested in devspace with old delegate

**Comment by abhinav-harness** [comment]
trigger uts

**Comment by abhinav-harness** [comment]
trigger uts

**Comment by abhinav-harness** [comment]
trigger uts

File: 970-ng-commons/src/main/java/io/harness/code/HarnessCodePayload.java [hunk 6]
```diff
@@ -28,10 +27,25 @@ public class HarnessCodePayload {
...
+  @Value
+  @Builder
+  public static class HarnessStagePayloadData implements HarnessPayloadData {
+    String account_id;
+    String org_identifier;
+    String project_identifier;
+    String pipeline_identifier;
+    String stage_identifier;
+    String execution_id;
+    String stage_id;
```
Comment (johannesHarness) [inline 970-ng-commons/src/main/java/io/harness/code/HarnessCodePayload.java:44]: what's stage_id vs stage_identifier? (given that harness is inconsistent with id/identifier and they are used interchangeably, I'm not sure if it's a duplicate entry, or actually meant to contain different data? if so, can we rename one of them?)

Comment (johannesHarness) [reply 970-ng-commons/src/main/java/io/harness/code/HarnessCodePayload.java:44]: should probably be `stage_identifier` and `stage_setup_id` based on the remaining code.

Comment (abhinav-harness) [reply 970-ng-commons/src/main/java/io/harness/code/HarnessCodePayload.java:44]: worked wrt https://harness.slack.com/archives/C0667TX5Y4Q/p1701471255205629

Comment (abhinav-harness) [reply 970-ng-commons/src/main/java/io/harness/code/HarnessCodePayload.java:44]: https://app.harness.io/ng/account/vpCkHKsDSxK9_KYfjCTMKA/module/ci/orgs/HarnessHCRInternalUAT/projects/Harness_Code/pipelines/stocoreprchecks/executions/inkPh62RTTKZksxwE0WVaQ/pipeline?stage=407qlC7DRXWlpafdd-QHtg&stageExecId=KAgWOo6uRSW2XuAjDpAkvA"

Stage id is from pipeline setup time which is `TS_tests` if you see in msgs

**Pull request merged**